# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: 2cf5cad90099
# Full Hash: 2cf5cad900992236d596bfb329099742ff15f689
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-20 04:08:16
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 4: Make ContentParent KeepAlives explicit with RAII references, r=smaug,dom-worker-reviewers,asuth
#   
#   This is a fairly significant patch, however it would be difficult to break it
#   down into smaller patches:
#   
# ==============================================================================

diff -r d920c2d72d00 -r 2cf5cad90099 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Wed Jun 19 20:14:49 2024 +0000
+++ b/dom/ipc/BrowserParent.h	Wed Jun 19 20:14:50 2024 +0000
@@ -19,6 +19,7 @@
 #include "mozilla/dom/BrowserBridgeParent.h"
 #include "mozilla/dom/PBrowserParent.h"
 #include "mozilla/dom/TabContext.h"
+#include "mozilla/dom/UniqueContentParentKeepAlive.h"
 #include "mozilla/dom/VsyncParent.h"
 #include "mozilla/dom/ipc/IdType.h"
 #include "mozilla/layout/RemoteLayerTreeOwner.h"
@@ -872,6 +873,12 @@
   // non-null.
   BrowserHost* mBrowserHost;
 
+  // KeepAlive for the containing process.
+  // NOTE: While this is a strong reference to ContentParent, which is
+  // cycle-collected, it is cleared as the BrowserParent's IPC connection is
+  // destroyed, so does not need to be cycle-collected.
+  UniqueContentParentKeepAlive mContentParentKeepAlive;
+
   ContentCacheInParent mContentCache;
 
   layout::RemoteLayerTreeOwner mRemoteLayerTreeOwner;